csharp/extractor/Semmle.Extraction.CSharp.Standalone/BuildAnalysis.cs

```diff
@@ -304,33 +315,17 @@ private void AnalyseProject(FileInfo project)
 
         }
 
-        private void Restore(string projectOrSolution)
+        private void Restore(string target)
         {
-            int exit;
-            try
-            {
-                exit = DotNet.RestoreToDirectory(projectOrSolution, packageDirectory.DirInfo.FullName);
-            }
-            catch (FileNotFoundException)
-            {
-                exit = 2;
-            }
-
-            switch (exit)
-            {
-                case 0:
-                case 1:
-                    // No errors
-                    break;
-                default:
-                    progressMonitor.CommandFailed("dotnet", $"restore \"{projectOrSolution}\"", exit);
-                    break;
-            }
+            dotnet.RestoreToDirectory(target, packageDirectory.DirInfo.FullName);
         }
 
-        private void RestoreSolutions(IEnumerable<string> solutions)
+        private void Restore(IEnumerable<string> targets)
         {
-            Parallel.ForEach(solutions, new ParallelOptions { MaxDegreeOfParallelism = 4 }, Restore);
+            foreach (var target in targets)
```

Comment:
Is restoring in parallel prone to transient errors?

Reply:
I disabled parallel execution, because the logs of the parallel `dotnet restore` calls overlap, and therefore it is difficult to diagnose issues.
`dotnet restore` itself is parallel, at least when it works on `*.sln` files.

Reply:
Maybe use .ForEach instead.